inference/generativeai/llm-workshop/lab4-openchatkit/deploy_openchatkit_on_sagemaker_deepspeed.ipynb

```diff
@@ -0,0 +1,1141 @@
+   "id": "3884d357",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "inference_image_uri = (\n",
+    "    f\"763104351884.dkr.ecr.{region}.amazonaws.com/djl-inference:0.21.0-deepspeed0.8.0-cu117\"\n",
+    ")\n",
+    "\n",
+    "print(f\"Image going to be used is ---- > {inference_image_uri}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4ddce346",
+   "metadata": {},
+   "source": [
+    "**Create the Tarball and then upload to S3 location**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c9c52338",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "!rm model.tar.gz\n",
+    "!tar czvf model.tar.gz -C code_gpt_neoxt-chatbase_deepspeed ."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f388dd32",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "s3_code_artifact = sess.upload_data(\"model.tar.gz\", bucket, s3_code_prefix)\n",
+    "print(f\"S3 Code or Model tar ball uploaded to --- > {s3_code_artifact}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e60ecd16",
+   "metadata": {},
+   "source": [
+    "### To create the end point the steps are:\n",
+    "\n",
+    "1. Create the Model using the Image container and the Model Tarball uploaded earlier\n",
+    "2. Create the endpoint config using the following key parameters\n",
+    "\n",
+    "    a) Instance Type is ml.g5.24xlarge \n",
+    "    \n",
+    "    b) ContainerStartupHealthCheckTimeoutInSeconds is 2400 to ensure health check starts after the model is ready    \n",
+    "3. Create the end point using the endpoint config created    \n",
+    "    "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "649cdd53",
+   "metadata": {},
+   "source": [
+    "#### Create the Model\n",
+    "Use the image URI for the DJL container and the s3 location to which the tarball was uploaded.\n",
+    "\n",
+    "The container downloads the model into the `/tmp` space on the instance because SageMaker maps the `/tmp` to the Amazon Elastic Block Store (Amazon EBS) volume that is mounted when we specify the endpoint creation parameter VolumeSizeInGB. It leverages `s5cmd`(https://github.com/peak/s5cmd) which offers a very fast download speed and hence extremely useful when downloading large models.\n",
+    "\n",
+    "For instances like p4dn, which come pre-built with the volume instance, we can continue to leverage the `/tmp` on the container. The size of this mount is large enough to hold the model.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "72f58be6",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "from sagemaker.utils import name_from_base\n",
+    "\n",
+    "model_name = name_from_base(f\"gpt-neoxt-chatbase-ds\")\n",
+    "print(model_name)\n",
+    "\n",
+    "create_model_response = sm_client.create_model(\n",
```

Comment:
for what reason why we don't use PYSDK? https://docs.djl.ai/docs/demos/aws/sagemaker/large-model-inference/standard_template_with_LMI_solution.html#42-create-sagemaker-endpoint